src/main/java/de/caritas/cob/userservice/filter/SubdomainExtractor.java

```diff
@@ -0,0 +1,35 @@
+package de.caritas.cob.userservice.filter;
+
+import com.google.common.net.InternetDomainName;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+@Component
+@Slf4j
+public class SubdomainExtractor {
```

Comment:
I suggest to use SubdomainExtractor from AgencyService to keep implementation consistent

Reply:
makes sense. I chaged it. since we did not have any review for this implemention (subdomain extractor topic) i would like to ask you does it makes sense to removeHttpPrefix in both classes instead to have this few lines in a seperate class. otherwise we have always to copy it. i also think that you could put everything in one class inseated of having it in 3, since we have to move around this class in other services.